com.unity.render-pipelines.universal/Runtime/ShaderDebugPrintManager.cs

```diff
@@ -0,0 +1,298 @@
+                                break;
+                            case DebugValueType.TypeFloat4:
+                                float4 valueFloat4 = *(float4*)&ptr[i + 1];
+                                outputLine += valueFloat4;
+                                i += 5;
+                                break;
+                            case DebugValueType.TypeBool:
+                                outputLine += ((data[i + 1] == 0) ? "False" : "True");
+                                i += 2;
+                                break;
+                            default:
+                                i = (int)count;  // Cannot handle the rest if there is an unknown type
+                                break;
+                        }
+
+                        outputLine += " ";
+                    }
+                }
+
+                if (count > 0)
+                {
+                    m_OutputLine = outputLine;
+                    m_OutputAction(outputLine);
+                }
+            }
+            else
+            {
+                const string errorMsg = "Error at read back!";
+                m_OutputLine = errorMsg;
+                m_OutputAction(errorMsg);
+            }
+        }
+
+        public void EndFrame()
+        {
+            int index = m_FrameCounter % FramesInFlight;
+            m_readbackRequests[index] = Rendering.AsyncGPUReadback.Request(m_OutputBuffers[index], m_bufferReadCompleteAction);
+
+            m_FrameCounter++;
+            m_FrameCleared = false;
+        }
+
+        public string OutputLine { get => m_OutputLine; }
+        public Action<string> OutputAction { set => m_OutputAction = value; }
+        public void DefaultOutput(string line)
+        {
+            Debug.Log(line);
+        }
+    }
+
+    public struct ShaderDebugPrintInput
+    {
+        // Mouse input
+        // GameView bottom-left == (0,0) top-right == (surface.width, surface.height) where surface == game display surface/rendertarget
+        // For screen pixel coordinates, game-view should be set to "Free Aspect".
+        // Works only in PlayMode
+        public Vector2 Pos { get; set; }
+        public bool LeftDown { get; set; }
+        public bool RightDown { get; set; }
+        public bool MiddleDown { get; set; }
+
+        public string String()
+        {
+            return $"Mouse: {Pos.x}x{Pos.y}  Btns: Left:{LeftDown} Right:{RightDown} Middle:{MiddleDown} ";
+        }
+    }
+
+    // TODO: Move, rename, perhaps #ifdef. Has dependencies to Input systems.
+    public static class ShaderDebugPrintInputProducer
+    {
+        static public ShaderDebugPrintInput Get()
+        {
+            var r = new ShaderDebugPrintInput();
+#if ENABLE_LEGACY_INPUT_MANAGER
+            r.Pos = Input.mousePosition;
+            r.LeftDown = Input.GetAxis("Fire1") > 0.5f;
+            r.RightDown = Input.GetAxis("Fire2") > 0.5f;
+            r.MiddleDown = Input.GetAxis("Fire3") > 0.5f;
```

Comment:
These lines should be using `Input.GetMouseButtonDown` or `Input.GetMouseButton` depending on if you need hold or trigger feedback. The reason is that these names Fire1/Fire2/Fire3 are defined in the projects input mapping, and these can often be removed or changed.

Reply:
fixed.